library/yum_versionlock.py

```diff
@@ -15,117 +15,265 @@
+            - 'package' accept wildcard format.
+        version_added: "2.7"
+    repo_mgr:
+        description:
+            - define the package manager to use to run the versionlock commands
+        default: based on the name of the module name (yum_versionlock or dnf_versionlock)
+        version_added: "2.10"
 # informational: requirements for nodes
+notes:
+    - Since version 2.10 the package is only require with the state: present/exclude/absent.
+    - The version 2.10 includes new states:
+    -- state: exclude - This prevent packages to be installed on the server
+    -- state: clear - This will clean the versionlock database
+    - The module will also return result from --check ansible option
 requirements:
-    - yum
-    - yum-versionlock
+    - yum / dnf
+    - yum-versionlock / dnf-versionlock
 author:
     - Florian Paul Hoberg <florian.hoberg@credativ.de>
 '''
 EXAMPLES = '''
+- name: Clear the dnf versionlock database
+  yum_versionlock:
+    state: clear
+    repo_mgr: dnf
 - name: Prevent Apache / httpd from beeing updated
   yum_versionlock:
     state: present
     package: httpd
+- name: Prevent nginx from beeing installed/updated
+  yum_versionlock:
+    state: exclude
+    package: nginx
+- name: Ensure python3* packages are not locked/excluded
+  yum_versionlock:
+    state: absent
+    package: python3*
 '''
 
-import os.path
+# Import 'os.path' to check versionlock config file
+# Import 're' to do search and strings substitution
+import os.path, re
 from ansible.module_utils.basic import AnsibleModule
 
-YUM_BINARY = "/bin/yum"
-
 
-def get_state_yum_versionlock():
-    """ Check for yum plugin dependency """
-    state = os.path.exists("/etc/yum/pluginconf.d/versionlock.conf")
+def get_state_versionlock(module, repo_cfg):
+    """ Check for plugin dependency """
+    state = os.path.exists(repo_cfg)
+    config_file = open(repo_cfg,"r")
+    for line in config_file.readlines():
+        if "enabled" in line:
+            enabled_state = line.split()[2]
+    if enabled_state != '1' and enabled_state != 'True':
+        module.warn("Warn: Plugin is installed but disabled ("+enabled_state+") in "+repo_cfg)
+    config_file.close
     return state
 
 
-def get_versionlock_packages(module):
-    """ Get an overview of all packages on yum versionlock """
-    rc_code, out, err = module.run_command("%s -q versionlock list"
-                                           % (YUM_BINARY))
+def get_packages(module, repo_mgr, list_type, package):
+    """ Get an overview of all packages available/installed """
+    rc_code, out, err = module.run_command("/usr/bin/%s -q list %s %s"
+                                           % (repo_mgr, list_type, package))
     if rc_code is 0:
-        return out
+        return out.splitlines()
     else:
-        module.fail_json(msg="Error: " + str(err) + str(out))
+        if rc_code == 1 and str(err) == 'Error: No matching Packages to list\n':
```

Comment:
On non us locales:
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "Unable to collect yum list installed : Fehler: Keine übereinstimmenden Pakete zum Auflisten\n - "}

Reply:
Fixed in commit 4f0d34c290a259a437fa9b7605093d0faa6aeb41